Reworks mutation testing docs (#271)
pestphp/docs: mutation-testing.md

The way mutation testing works is by introducing small changes (mutations) to the source code and verifying if the tests are failing against these changes. This process helps developers identify weak spots in their test suites and improve the overall quality of their tests.

Unlike with other mutation testing libraries, our plugin is deeply integrated into Pest PHP. So, each time a mutation is introduced, Pest PHP will:

- **Only run the tests covering the mutated code** to speed up the process.
- **Cache as much as possible** to speed up the process on subsequent runs.
- If enabled, use **parallel execution to run multiple tests** in parallel to speed up the process.

To get started with mutation testing, head over to your test file, and be specific about which part of your code your test covers using the `covers()` function or the `covers()` method.

```php
<?php

covers(TodoController::class);

it('list todos', function () {
    //
});

it('create todos', function () {
    //
})->covers(CreateTodoRequest::class);
```

Finally, run Pest PHP with the `--mutate` option to start mutation testing.

```bash
./vendor/bin/pest --mutate
```

> Optionally, you may run mutation testing in parallel to speed up the process. Of course, your test suite must be able to run in tests in parallel.

Reviewer: ```suggestion
Optionally, you may run mutation testing in parallel to speed up the process. Of course, your test suite must be able to run the tests in parallel.
```

```bash
./vendor/bin/pest --mutate --parallel
```

> After running mutation testing, you will see a detailed report showing the number of mutations, how many were caught, and the mutation score.

Reviewer: ```suggestion
After running mutation testing, you will see a detailed report showing the number of mutations, how many were untested, and the mutation score.
```

```
[IMAGE HERE]
```

The higher the mutation score, the better your test suite is. A mutation score of 100% means that all mutations were "covered", which is the goal of mutation testing.

> Now, if you see "uncovered" mutations, or are a mutation score below 100%, typically means that you have **missing tests** or that **your tests are not covering all the edge cases**.

Reviewer: ```suggestion
Now, if you see "untested" mutations, or have a mutation score below 100%, typically means that you have **missing tests** or that **your tests are not covering all the edge cases**.
```

> <a name="covered-vs-uncovered-mutations"></a>
> ## Covered Vs Uncovered Mutations
> 
> When running mutation testing, you will see two types of mutations: **covered** and **uncovered** mutations.
> 
> - **Covered Mutations**: These are mutations that were detected by your test suite. They are considered "covered" because your tests were able to catch the changes introduced by the mutation.
> 
> As example, the following mutation is considered "covered" because the test suite was able to detect the change.

Reviewer: ```suggestion
<a name="tested-vs-untested-mutations"></a>
## Tested vs Untested and Uncovered Mutations

When running mutation testing, you will see three types of mutations: **tested**, **untested** and **uncovered** mutations.

- **Tested Mutations**: These are mutations that were detected by your test suite. They are considered "tested" because your tests were able to catch the changes introduced by the mutation.

As example, the following mutation is considered "tested" because the test suite was able to detect the change.
```

```diff
class TodoController
{
    public function index(): array
    {
-        return Todo::all()->toArray();
+        return [];
    }
}

it('list todos', function () {
    Todo::factory()->create(['name' => 'Buy milk']);

    // this fails because the mutation changed the return value, proving that the test is working and testing the return value...
    $this->getJson('/todos')->assertStatus(200)->assertJsonContains([
        ['name' => 'Buy milk'], 
    ]);
});
```

> - **Uncovered Mutations**: These are mutations that were not detected by your test suite. They are considered "uncovered" because your tests were not able to catch the changes introduced by the mutation.
> 
> As example, the following mutation is considered "uncovered" because the test suite was not able to detect the change.

Reviewer: ```suggestion
- **Untested Mutations**: These are mutations that were not detected by your test suite. They are considered "untested" because your tests were not able to catch the changes introduced by the mutation.

As example, the following mutation is considered "untested" because the test suite was not able to detect the change.
```

```diff
class TodoController
{
    public function index(): array
    {
-        return Todo::all()->toArray();
+        return [];
    }
}

it('list todos', function () {
    Todo::factory()->create(['name' => 'Buy milk']);

    // this test still passes even though the return value was changed by the mutation... 
    $this->getJson('/todos')->assertStatus(200);
});
```

Changing the return value is only one of many possible mutations. Typically, a mutation can be a change in the return value, a change in the method call, a change in the method arguments, and so on.

> 

Reviewer: ```suggestion
- **Uncovered Mutations**: These are mutations were no test is excuting the mutated code at all. In this case it **always** means that a test is missing.

```
